lib/ddtrace/context.rb

```diff
@@ -50,6 +50,7 @@ def sampling_priority
 
     def sampling_priority=(priority)
       @mutex.synchronize do
+        @sampled = true
```

Comment:
This does not work, for 2 reasons:

* I think it would put `sampled` to true even if the arg (`priority`) is `nil` or `0`
* in all cases it breaks the stats, because it alters the numbers of traces we send without changing the `sample_rate` metric (from which we deduce the weight). In short, it breaks the stats.